packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -237,6 +242,47 @@ export class InteractionTracker {
     stopWatch.stopAndRecord();
     delete this.interactions[interactionId];
   }
+
+  // Monkey-patch `History API` to detect route transitions.
+  // This is necessary because there might be some cases when
+  // there are several interactions being tracked at the same time
+  // but if there is an user interaction that triggers a route transition
+  // while those interactions are still in tracking, only that interaction
+  // will have a `Navigation` name. Otherwise, if this is not patched, the
+  // other interactions will change the name to `Navigation` even if they
+  // did not cause the route transition.
+  private patchHistoryApi() {
+    const pushState = history.pushState;
+    history.pushState = (
+      //tslint:disable:no-any
+      data: any,
+      title: string,
+      url?: string | null | undefined
+    ) => {
+      pushState.call(history, data, title, url);
+      this.maybeUpdateInteractionName();
+    };
+
+    const replaceState = history.replaceState;
+    history.replaceState = (
+      //tslint:disable:no-any
+      data: any,
+      title: string,
+      url?: string | null | undefined
+    ) => {
+      replaceState.call(history, data, title, url);
+      this.maybeUpdateInteractionName();
+    };
+  }
+
+  private maybeUpdateInteractionName() {
+    console.log('location changed!');
+    // Route transition is detected, thus the current interaction name might change.
+    const rootSpan = getRootSpan(Zone.current);
+    if (rootSpan && rootSpan.name.startsWith('<')) {
+      rootSpan.name = 'Navigation ' + location.pathname;
```

Comment:
Would it make sense to double check that `location.pathname` actually changed since the interaction started (you are still capturing it above, though as an `href` so I think if you just changed that to ? That way you can double check that the user didn't call `replaceState` and just ended up going to a URL they were already at.

I also think that doing `${location.pathname}#${location.hash}` (and maybe putting that in a function) would be good because sometimes people do routing just based on the hash in the URL.

Reply:
Folowing [the browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/History#Browser_compatibility) I see it is not necessary to change the name using the hash as it does not actually do a route transition.

Also, chatting offline, we found valuable to store the `location.pathname` in the transition.